### PR TITLE
Add TEXTJOIN function with i18n and documentation

### PR DESCRIPTION
**Pull request by marcin-kordas-hoc**
## Summary

Adds the `TEXTJOIN` function — joins text from multiple strings and/or ranges with a configurable delimiter.

Replaces #1625 (was opened from fork, now from upstream branch directly).

### Features
- Scalar and array/range delimiter support with cycling behavior
- `ignore_empty` parameter to skip empty strings
- Type coercion (numbers, booleans → strings)
- Error propagation from both delimiter and text arguments
- 32,767 character limit (Excel compatibility)
- i18n translations for all 17 supported languages
- Documentation in `built-in-functions.md`

### Implementation
- New `textjoin` method + `flattenArgToStrings` helper in `TextPlugin`
- `repeatLastArgs: 1` metadata pattern (same as SUMPRODUCT, etc.)
- Defensive `CellError` check on `coerceScalarToString` return value

### Changed files
| File | Change |
|------|--------|
| `src/interpreter/plugin/TextPlugin.ts` | `textjoin()` + `flattenArgToStrings()` |
| `src/error-message.ts` | `TextJoinResultTooLong` message |
| `src/i18n/languages/*.ts` (17 files) | TEXTJOIN translations |
| `docs/guide/built-in-functions.md` | TEXTJOIN row (alphabetically between TEXT and TRIM) |

### Review feedback addressed (from #1625)
- Tests moved to private `hyperformula-tests` repo (companion PR pending)
- Fixed docs alphabetical ordering
- Fixed unsafe `as string` cast in `flattenArgToStrings`

## Test plan

- [x] 35 tests in `hyperformula-tests/unit/interpreter/function-textjoin.spec.ts`
- [x] Full suite: 480 suites / 5396 tests passed

---

> [!NOTE]
> **Medium Risk**
> Adds a new interpreter function (`TEXTJOIN`) with range flattening and type coercion, which touches formula evaluation paths and may introduce edge-case regressions around error propagation and large-string handling.
> 
> **Overview**
> Adds the new `TEXTJOIN` spreadsheet function, including interpreter support for joining scalars and ranges with a delimiter (including delimiter cycling), optional skipping of empty strings, and consistent error propagation.
> 
> Introduces a new `ErrorMessage.ResultTooLong` and enforces Excel’s 32,767-character output limit (returning `#VALUE!` when exceeded).
> 
> Updates function documentation, the unreleased changelog, and adds `TEXTJOIN` translations across all supported language packs.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 81426d7da682abb5d5cbf9f1c989125777d07abb. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/interpreter/plugin/TextPlugin.ts [hunk 1]
```diff
@@ -424,6 +434,78 @@ export class TextPlugin extends FunctionPlugin implements FunctionPluginTypechec
...
+        }
+
+        const parts = ignoreEmpty ? texts.filter((t) => t !== '') : texts
+
+        if (parts.length === 0) {
+          return ''
+        }
+        let result = parts[0]
+        for (let i = 1; i < parts.length; i++) {
+          result += delimiters[(i - 1) % delimiters.length] + parts[i]
```
[the remark this answers is not in the document]
Comment (sequba) [reply src/interpreter/plugin/TextPlugin.ts]: Please add a test that verifies the behavior of `TEXTJOIN` with an empty delimiter and empty array provided as delimiter. E.g.:
`=TEXTJOIN("", TRUE(), "Hello", "World")`,
`=TEXTJOIN(, TRUE(), "Hello", "World")`,
`=TEXTJOIN({}, TRUE(), "Hello", "World")`

File: src/interpreter/plugin/TextPlugin.ts [hunk 2]
```diff
@@ -424,6 +434,78 @@ export class TextPlugin extends FunctionPlugin implements FunctionPluginTypechec
...
+  private flattenArgToStrings(arg: InternalScalarValue | SimpleRangeValue): string[] | CellError {
+    const values = arg instanceof SimpleRangeValue ? arg.valuesFromTopLeftCorner() : [arg]
+    const result: string[] = []
+    for (const val of values) {
+      if (val instanceof CellError) {
+        return val
+      }
+      const coerced = coerceScalarToString(val as InternalScalarValue)
+      if (coerced instanceof CellError) {
+        return coerced
```
Comment (sequba) [inline src/interpreter/plugin/TextPlugin.ts:502]: Codecov reports this line is not covered by tests. Is it possible to add a test for this case?

**Review by sequba — CHANGES_REQUESTED** [review]
Good implementation. There are only minor issues. Also, please, add a CHANGELOG entry.

File: src/interpreter/plugin/TextPlugin.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcin-kordas-hoc) [reply src/interpreter/plugin/TextPlugin.ts]: Empty delimiters array is unreachable: `flattenArgToStrings` processes either a scalar (always produces `[""]` for empty) or a range (`valuesFromTopLeftCorner()` always returns at least one cell since `AbsoluteCellRange` width/height are always >= 1). Empty array literal `{}` causes a parse error before reaching this code.

[the remark this answers is not in the document]
Comment (marcin-kordas-hoc) [reply src/interpreter/plugin/TextPlugin.ts]: Added 3 tests in hyperformula-tests (`db1ef4c`):
- `=TEXTJOIN("", TRUE(), "Hello", "World")` → `"HelloWorld"`
- `=TEXTJOIN(, TRUE(), "Hello", "World")` → `"HelloWorld"` (omitted delimiter = empty string)
- `=TEXTJOIN({}, TRUE(), "Hello", "World")` → parse error (`{}` is not valid syntax)

File: src/interpreter/plugin/TextPlugin.ts (hunk 2 above)
Comment (marcin-kordas-hoc) [reply src/interpreter/plugin/TextPlugin.ts:502]: This branch is unreachable: `val instanceof CellError` on the line above catches all CellError values before `coerceScalarToString` is called, and `coerceScalarToString` only returns CellError when given a CellError input. It handles string, EmptyValue, ExtendedNumber, and boolean — all of which return strings.

Kept as a defensive guard in case `coerceScalarToString` is ever extended to error on new value types.

File: src/interpreter/plugin/TextPlugin.ts [hunk 3]
```diff
@@ -424,6 +434,79 @@ export class TextPlugin extends FunctionPlugin implements FunctionPluginTypechec
...
+    return values.reduce<string[] | CellError>((acc, val) => {
+      if (acc instanceof CellError) {
+        return acc
+      }
+      if (val instanceof CellError) {
+        return val
+      }
+      const coerced = coerceScalarToString(val as InternalScalarValue)
+      return coerced instanceof CellError ? coerced : [...acc, coerced]
+    }, [])
```
[the remark this answers is not in the document]
Comment (marcin-kordas-hoc) [reply src/interpreter/plugin/TextPlugin.ts]: Fixed in `6d563e5c` — reverted to imperative loops with `push()` which are O(N). The `reduce` + spread pattern was indeed O(N²).

File: src/interpreter/plugin/TextPlugin.ts [hunk 4]
```diff
@@ -424,6 +434,78 @@ export class TextPlugin extends FunctionPlugin implements FunctionPluginTypechec
...
+  private flattenArgToStrings(arg: InternalScalarValue | SimpleRangeValue): string[] | CellError {
+    const values = arg instanceof SimpleRangeValue ? arg.valuesFromTopLeftCorner() : [arg]
+    return values.reduce<string[] | CellError>((acc, val) => {
+      if (acc instanceof CellError) {
+        return acc
+      }
+      if (val instanceof CellError) {
+        return val
+      }
+      return [...acc, coerceScalarToString(val as InternalScalarValue) as string]
```
[the remark this answers is not in the document]
Comment (marcin-kordas-hoc) [reply src/interpreter/plugin/TextPlugin.ts]: Fixed in `6d563e5c` — the `as string` cast is removed. The defensive `coerceScalarToString` error check is restored as a proper `instanceof CellError` guard.

File: docs/guide/custom-functions.md [hunk 5]
```diff
@@ -358,18 +358,12 @@ it('returns a VALUE error if the range argument contains a string', () => {
 
 ## Working demo
 
+Explore the full working example on [Stackblitz](https://stackblitz.com/github/handsontable/hyperformula-demos/tree/3.2.x/custom-functions?v=${$page.buildDateURIEncoded}).
```
[the remark this answers is not in the document]
Comment (marcin-kordas-hoc) [reply docs/guide/custom-functions.md:361]: Pre-existing issue, not related to TEXTJOIN changes. The markdown template literal interpolation is outside the scope of this PR.

File: src/interpreter/plugin/TextPlugin.ts (hunk 2 above)
Comment (sequba) [reply src/interpreter/plugin/TextPlugin.ts:502]: ok, gotcha

**Review by sequba — APPROVED** [review]
